.taskcluster.yml

```diff
@@ -0,0 +1,50 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+version: 0
+allowPullRequests: public
+tasks:
+###############################################################################
+# Task: Pull requests and pushes
+#
+# Triggered whenever a pull request is opened or updated, or a push is being made
+#
+# - Build the library
+# - Run unit tests
+# - Run code quality tools
+###############################################################################
+  - provisionerId: '{{ taskcluster.docker.provisionerId }}'
+    workerType: '{{ taskcluster.docker.workerType }}'
+    extra:
+      github:
+        events:
+          - pull_request.opened
+          - pull_request.edited
+          - pull_request.synchronize
+          - pull_request.reopened
+          - push
+    payload:
+      maxRunTime: 3600
+      deadline: "{{ '2 hours' | $fromNow }}"
+      image: 'mozillamobile/fretboard'
+      command:
+        - /bin/bash
+        - '--login'
+        - '-cx'
+        - >-
+          git fetch {{ event.head.repo.url }} {{ event.head.repo.branch }}
+          && git config advice.detachedHead false
+          && git checkout {{event.head.sha}}
+          && ./gradlew --no-daemon clean assemble test detektCheck ktlint lint jacocoTestReport
+          && bash <(curl -s https://codecov.io/bash)
```

Comment:
If you are re-using the android-components image then you will need to create a new checkout of this repository first. For example:
```bash
mkdir /opt/fretboard
&& cd /opt/fretboard
&& git clone {{ event.head.repo.url }}
&& git config advice.detachedHead false
&& git checkout {{event.head.sha}}
[..]
```